cohortextractor/process_covariate_definitions.py

```diff
@@ -382,6 +380,60 @@ def type_of_aggregate_of(self, column_names, aggregate_function, **kwargs):
         return column_type
 
 
+def process_date_expressions(covariate_definitions, index_date):
+    output = {}
+    for name, (query_type, query_args) in covariate_definitions.items():
+        for key in ("reference_date", "start_date", "end_date"):
+            if key in query_args:
+                query_args[key] = process_date_expression(query_args[key], index_date)
+        if "between" in query_args:
+            start, end = query_args["between"]
+            query_args["between"] = (
+                process_date_expression(start, index_date),
+                process_date_expression(end, index_date),
+            )
+        if "return_expectations" in query_args:
+            query_args["return_expectations"] = process_expectations_definition(
+                query_args["return_expectations"], index_date
+            )
+        output[name] = (query_type, query_args)
+    return output
+
+
+def process_expectations_definition(expectations_definition, index_date):
```

Comment:
I wonder if this should be `process_date_expressions_in_expectations_definition` or similar, to make it clearer what's happening when it's called?